library/src/test/java/com/chuckerteam/chucker/api/ChuckerInterceptorTest.kt

```diff
@@ -198,4 +198,86 @@ class ChuckerInterceptorTest {
 
         assertThat(responseBody).isNull()
     }
+
+    @ParameterizedTest
+    @EnumSource(value = ClientFactory::class)
+    fun payloadSize_dependsOnTheAmountOfDataDownloaded(factory: ClientFactory) {
+        val segmentSize = 8_192
+        val body = Buffer().apply {
+            repeat(10 * segmentSize) { writeUtf8("!") }
+        }
+        server.enqueue(MockResponse().setBody(body))
+        val request = Request.Builder().url(serverUrl).build()
+
+        val client = factory.create(chuckerInterceptor)
+        val source = client.newCall(request).execute().body()!!.source()
+        source.use { it.readByteString(segmentSize.toLong()) }
+
+        val transaction = chuckerInterceptor.expectTransaction()
+        // We cannot expect exact amount of data as there are no guarantees that client
+        // will read from the source exact amount of data that we requested.
+        //
+        // It is only best effort attempt and if we download less than 8KiB reading will continue
+        // in 8KiB batches until at least 8KiB is downloaded.
+        assertThat(transaction.responsePayloadSize).isAtMost(2 * segmentSize)
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = ClientFactory::class)
+    fun contentLength_isProperlyReadFromHeader_andNotFromAmountOfData(factory: ClientFactory) {
+        val segmentSize = 8_192
+        val body = Buffer().apply {
+            repeat(10 * segmentSize) { writeUtf8("!") }
+        }
+        server.enqueue(MockResponse().setBody(body))
+        val request = Request.Builder().url(serverUrl).build()
+
+        val client = factory.create(chuckerInterceptor)
+        val source = client.newCall(request).execute().body()!!.source()
+        source.use { it.readByteString(segmentSize.toLong()) }
+
+        val transaction = chuckerInterceptor.expectTransaction()
+        assertThat(transaction.responseHeaders).contains(
+            """
+            |  {
+            |    "name": "Content-Length",
+            |    "value": "${body.size()}"
+            |  }
+            """.trimMargin()
```

Comment:
You should be able to remove the `|` and change this to `.trimIndent()`

Reply:
It won't work, unfortunately. JSON has this structure.

```json
[
  {
    "name": "Content-Length",
    "value": "x"
  }
]
```

`.trimIndent()` would remove first indentation before `{` and `}`. And I didn't want to compare the whole content as it would make assumptions on no other headers being there.